app/src/test/java/org/mozilla/focus/browser/WebViewCacheTest.kt

```diff
@@ -0,0 +1,46 @@
+package org.mozilla.focus.browser
+
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import mozilla.components.browser.engine.system.SystemEngineView
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.mock
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+@RunWith(RobolectricTestRunner::class)
+class WebViewCacheTest {
+
+    @Test
+    fun `WHEN getWebView is called multiple times THEN the same WebView is returned`() {
+        val cachRefs = List(5) { getWebView() }
+        cachRefs.forEach { assertEquals(cachRefs.first(), it) }
+    }
+
+    @Test
+    fun `GIVEN cached WebView has a parent WHEN WebView is returned THEN WebView should be removed from parent`() {
+        for (i in 0..5) {
+            val webView = getWebView()
+            assertEquals(null, webView.parent)
+            val parent = FrameLayout(RuntimeEnvironment.application)
```

Comment:
nit: If we do set-up inside the method, I'd use newlines to separate setup (GIVEN) from WHEN and THEN, unless the function is trivial. In this case, I think GIVEN should be separated but WHEN and THEN can probably be put together.